src/uharfbuzz/_harfbuzz.pyx

```diff
@@ -409,6 +409,11 @@ cdef class Font:
         cdef Font inst = cls(face)
         return inst
 
+    def create_sub_font(self, parent: Font):
+        cdef Font inst = self.__class__(self.face)
```

Comment:
i think this leads to a memory leak because we allocate a new inst._hb_font from the same self.face (self.__class__() uses the default constructor, hence it calls `__cinit__`), then immediately afterwards we are discarding that and setting inst._hb_font without freeing the memory (as we it's done in `__dealloc__`).

I think the current approach for alternate constructors is to skip `__cinit__` altogether and use `__new__` instead to get an instance of the class with fields still not initialized and initialize those explicitly like a constructor would do. 

```cython
    def create_sub_font(self):
        cdef Font inst = self.__class__.__new__(self.__class__)
        inst._hb_font = hb_font_create_sub_font(self._hb_font)
        inst._face = self._face
        return inst
```

Reply:
Done.